conans/model/conan_file.py

```diff
@@ -403,48 +400,6 @@ def generators_path(self) -> Path:
         assert self.generators_folder is not None, "`generators_folder` is `None`"
         return Path(self.generators_folder)
 
-    def source(self):
-        pass
-
-    def system_requirements(self):
-        """ this method can be overwritten to implement logic for system package
-        managers, as apt-get
-        """
-
-    def config_options(self):
-        """
-        Modify options, probably conditioned to some settings. This call is executed
-        before config_settings. E.g.
-        if self.settings.os == 'Windows':
-        del self.options.shared  # shared/static not supported in win
-        """
-
-    def configure(self):
-        """
-        Modify settings, probably conditioned to some options. This call is executed
-        after config_options. E.g.
-        if self.options.header_only:
-        self.settings.clear()
-        This is also the place for conditional requirements
-        """
-
-    def build(self):
-        """
-        Build your project calling the desired build tools as done in the command line.
-        E.g. self.run("cmake --build .") Or use the provided build helpers. E.g. cmake.build()
-        """
-        self.output.warning("This conanfile has no build step")
-
-    def package(self):
-        """ package the needed files from source and build folders.
-        E.g. copy(self, "\*.h", os.path.join(self.source_folder, "src/includes"), os.path.join(self.package_folder, "includes"))
-        """
-        self.output.warning("This conanfile has no package step")
-
-    def package_info(self):
-        """ define cpp_build_info, flags, etc
-        """
```

Comment:
Should we add as ConanFile's docstring the list of methods? Only to be able to see the list of all the methods available.

Reply:
To the general docstring of the class? If you are reading the docstring in the conan docs, you will already have there the list of methods available, won't you?